includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -1471,6 +1471,30 @@ public function process_payment( $order_id ) {
 		return array();
 	}
 
+	/**
+	 * Update Checkout Session Data
+	 *
+	 * @param  string $checkout_session_id Checkout Session Id.
+	 * @param  array  $payload Data to send to the API.
+	 * @return object|WP_Error API Response, or WP_Error.
+	 */
+	protected function update_checkout_session_data( $checkout_session_id, $payload ) {
```

Comment:
This change is positive as the code gets more testable/mockable. I still will need 2nd iteration of the review to ensure it's the best approach.

Reply:
From data flow PoV would it be possible to add types declarations here: like `protected function update_checkout_session_data( string $checkout_session_id, array $payload )`?